yjit/src/codegen.rs

```diff
@@ -6347,6 +6347,42 @@ fn jit_thread_s_current(
     true
 }
 
+fn jit_numeric_dup(
+    _jit: &mut JITState,
+    _asm: &mut Assembler,
+    _ci: *const rb_callinfo,
+    _cme: *const rb_callable_method_entry_t,
+    _block: Option<BlockHandler>,
+    _argc: i32,
+    _known_recv_class: Option<VALUE>,
+) -> bool {
+    // Numeric#dup has arity=0 and is the identity function.
+    // Our caller already did argument count check, so this is
+    // no-op to return the receiver that is already on the stack.
+    true
```

Comment:
Could we propagate the type here if known/fixnum?

Reply:
Because of the guard on the receiver type info is already in the context and we don't need to do anything to propagate it here.